File: Frontend/job-portal-frontend/src/services/jobCategoryService.js
import api from "./api";

export const getCategories = () => {
    return api.get("/categories");
};

export const getCategoryById = (id) => {
    return api.get(`/categories/${id}`);
};

export const addCategory = (category) => {
    return api.post("/categories", category);
};

export const updateCategory = (id, category) => {
    return api.put(`/categories/${id}`, category);
};

export const deleteCategory = (id) => {
    return api.delete(`/categories/${id}`);
};

File: Backend/shivani-technologies-backend/src/main/java/com/shivani/service/JobCategoryService.java
package com.shivani.service;

import java.util.List;

import com.shivani.entity.JobCategory;

public interface JobCategoryService {

    JobCategory saveCategory(JobCategory category);

    List<JobCategory> getAllCategories();

    JobCategory getCategoryById(Long id);

    JobCategory updateCategory(Long id, JobCategory category);

    void deleteCategory(Long id);

}